src/components/questionnaire/questionnaire-table.tsx

```diff
@@ -0,0 +1,155 @@
+      setLoading(true);
+      try {
+        const questionnaires = await questionnaireApi.listQuestionnaires();
+        setQuestionnaires(questionnaires);
+      } finally {
+        setLoading(false);
+      }
+    };
+    fetchQuestionnaires();
+  }, []);
+
+  const handleOpenDialog = (id: string, title: string) => {
+    setDeleteId(id);
+    setDeleteTitle(title);
+    setDialogOpen(true);
+  };
+
+  const handleCloseDialog = () => {
+    setDialogOpen(false);
+    setDeleteId(null);
+    setDeleteTitle(null);
+  };
+
+  const handleConfirmDelete = async () => {
+    if (!deleteId) return;
+    setLoading(true);
+    try {
+      await questionnaireApi.deleteQuestionnaire({ id: deleteId });
+      setQuestionnaires((prevQuestionnaires: Questionnaire[]) =>
+        prevQuestionnaires.filter((questionnaire: Questionnaire) => questionnaire.id !== deleteId)
+      );
+      handleCloseDialog();
+    } catch (error) {
+      console.error("Delete Questionnaire failed:", error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const columns = [
+    { field: "title", headerName: `${strings.questionnaireTable.title}`, flex: 3 },
+    { field: "description", headerName: `${strings.questionnaireTable.description}`, flex: 5 },
+    adminMode
+      ? {
+          field: "actions",
+          headerName: `${strings.questionnaireTable.actions}`,
+          flex: 2.5,
+          renderCell: (params: GridRenderCellParams) => (
+            <>
+              <Button variant="outlined" color="success" sx={{ mr: 1 }}>
+                <EditIcon sx={{ color: "success.main", mr: 1 }} />
+                {strings.questionnaireTable.edit}
+              </Button>
+              <Button
+                variant="contained"
+                color="secondary"
+                onClick={() => handleOpenDialog(params.row.id, params.row.title)}
+              >
+                <DeleteForeverIcon sx={{ color: "red", mr: 1 }} />
+                {strings.questionnaireTable.delete}
+              </Button>
+            </>
+          )
+        }
+      : {
+          field: "status",
+          headerName: `${strings.questionnaireTable.status}`,
+          flex: 1,
+          renderCell: (params: GridRenderCellParams) =>
+            params.row.passedUsers && params.row.passedUsers.lenght > 0 ? (
+              <CheckCircleIcon sx={{ color: "green" }} />
+            ) : (
+              <CloseIcon sx={{ color: "red" }} />
+            )
+        }
+  ];
+
+  return (
+    <>
+      <Paper style={{ height: 500, width: "100%" }}>
+        {loading && (
+          <CircularProgress
+            size={50}
+            style={{
+              position: "absolute",
+              top: "50%",
+              left: "50%",
+              transform: "translate(-50%, -50%)"
+            }}
+          />
+        )}
+        <DataGrid
+          rows={questionnaires}
+          columns={columns}
+          pagination
+          pageSizeOptions={[5]}
```

Comment:
Would be nice to let the user customize this- should be built in option with the data Grid already